File: 0128-longest-consecutive-sequence/0128-longest-consecutive-sequence.cpp
class Solution {
public:
    int longestConsecutive(vector<int>& nums) {
        unordered_set<int>s(nums.begin(),nums.end());
        int longest=0;
        for(auto &x:s){
            if(!s.count(x-1)){
                int length=1;
                int curr=x;
                while(s.count(curr+1)){
                    curr++;
                    length++;
                }
                longest=max(longest,length);
            }
        }
        return longest;
    }
};

File: 0128-longest-consecutive-sequence/0128-longest-consecutive-sequence.py
class Solution:
    def longestConsecutive(self, nums: list[int]) -> int:
        s=set(nums)
        longest=0
        for x in s:
            if x-1 not in s:
                length=1
                curr=x
                while curr+1 in s:
                    curr+=1
                    length+=1
                longest=max(longest,length)
        return longest